repose-aggregator/components/services/http-client-service/http-client-service-impl/src/test/groovy/org/openrepose/core/services/httpclient/impl/HttpConnectionPoolProviderTest.groovy

```diff
@@ -107,8 +125,124 @@ class HttpConnectionPoolProviderTest {
     public void "should not add header parameter when not configured"() {
         poolType.setHeaders(null)
 
-        DefaultHttpClient client = HttpConnectionPoolProvider.genClient(poolType) as DefaultHttpClient
+        DefaultHttpClient client = HttpConnectionPoolProvider.genClient("", poolType) as DefaultHttpClient
 
         assert !client.getParams().getParameter(ClientPNames.DEFAULT_HEADERS)
     }
+
+    @Test
+    public void "should create a pool that can connect to a server with client auth using keystore and truststore when configured"() {
+        server = new Server()
+
+        // SSL Context Factory
+        def sslContextFactory = new SslContextFactory()
+        sslContextFactory.keyStoreResource = Resource.newClassPathResource("server.jks")
+        sslContextFactory.keyStorePassword = "password"
+        sslContextFactory.keyManagerPassword = "password"
+        sslContextFactory.needClientAuth = true
+        sslContextFactory.trustStoreResource = Resource.newClassPathResource("client.jks")
+        sslContextFactory.trustStorePassword = "password"
+
+        // SSL HTTP Configuration
+        def https_config = new HttpConfiguration()
+        https_config.addCustomizer new SecureRequestCustomizer()
+
+        // SSL Connector
+        def sslConnector = new ServerConnector(
+                server,
+                new SslConnectionFactory(sslContextFactory, HttpVersion.HTTP_1_1.asString()),
+                new HttpConnectionFactory(https_config)
+        )
+        sslConnector.setPort(0)
+
+        // Start the server
+        def statusCode = HttpServletResponse.SC_OK
+        def responseContent = "The is the plain text test body data.\n".bytes
+        def contentType = "text/plain;charset=utf-8"
```

Comment:
Considering you're always declaring the charset to be `UTF-8`, which is totally legit, you also probably want to get the `String` as `Byte`s in the previous line using that charset. That is, I believe `"string".bytes` uses the platform default character encoding, which is usually, but not necessarily, `UTF-8`. So:
`def responseContent = "The is the plain text test body data.\n".getBytes(StandardCharsets.UTF_8)`